src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,465 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Runtime.InteropServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// The MLPerf Training workload executor.
+    /// </summary>
+    [UnixCompatible]
+    public class MLPerfTrainingExecutor : VirtualClientComponent
+    {
+        private const string AccuracySummary = nameof(MLPerfTrainingExecutor.AccuracySummary);
+        private const string PerformanceSummary = nameof(MLPerfTrainingExecutor.PerformanceSummary);
+
+        private IStateManager stateManager;
+        private ISystemManagement systemManager;
+
+        /// <summary>
+        /// Constructor for <see cref="MLPerfTrainingExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public MLPerfTrainingExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+             : base(dependencies, parameters)
+        {
+            this.systemManager = this.Dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManager.StateManager;
+
+            if (string.IsNullOrEmpty(this.DiskFilter))
```

Comment:
Do we really need these 4 lines of code as at line 53, we are already setting default value if DiskFilter will be null or empty?

Reply:
Removed them.